iot/contiki/README.md

```diff
@@ -0,0 +1,162 @@
+First, we will go through an explanation of every section inside the `contiki.def` file:
+
+#### Environment section
+
+Here we define all the needed environment variables for ContikiOS to work fine.
+For this specific example the only needed variable is `CONTIKI` which is set to the
+path of the installation of contiki in the container.
+
+#### Help section
+
+Here we define the description of the container, in this case this container has
+ContikiOS as well as all the examples that come along including the Cooja simulator. We will test the functionality
+of this container by running a `Hello world` example below, by command line and by simulator.
+The help section also displays the correct way to run your container.
+
+#### Post section
+
+Here we define the needed dependencies to be installed in order to make ContikiOS run smoothly.
+One thing that is important is the set up of the dependencies for the specific architecture.
+The following example is based on a 64-bit architecture.
+
+Some other needed dependencies include the Java installation as well as the support for MSP430-based platforms and Java for Cooja.
+
+#### Runscript section
+
+Here all the needed commands to run the ContikiOS example are specified. More in detail, the Hello-world example provided from Contiki is run to know that the installation inside the container has been successful.
+
+
+How to build my Singularity ContikiOS container?
+================================================
+
+To build this container, you will need the following:
+
+ - Singularity, which you can download and install from [here](https://github.com/sylabs/singularity).
+ - A text editor, like: `micro`, `vim` or `nano`.
+ - Root access.
+
+#### Build the container
+
+We will build and save the container into the `/contiki` folder. Make sure you have this folder before following the example. Go inside this folder in your host:
+
+```
+$ cd /contiki
+```
+
+And then you can build the container with a single command like this:
+
+```
+$ sudo singularity build contiki.sif contiki.def
+```
+
+After this, you are ready to set up your container before running the example.
+
+#### Set up
+
+First make sure to create the following directory structure on your host:
+
+```
+/contiki
+|     |--examples
+|     |        \--hello-world
+|     \--tools
+```
+
+This is the code in C that the node will execute at the moment of using the Cooja simulator from our example inside the Singularity container.
+
+Another task to do before using the container is to copy certain code that resides inside the container and that will be later bound into your host. To do this, you will first shell into the container like so:
+
+```
+$ sudo singularity shell contiki.sif
+```
+
+Now that you are inside the container you can copy all the needed code for compilation:
+
+```
+Singularity contiki.sif:/contiki> cp -a /contiki/examples/hello-world/* ./examples/hello-world/
```

Comment:
Not necessary, but
I would change the prompt from: `Singularity contiki.sif:/contiki>` to `> `,